static/elements/chromedash-featurelist.js

```diff
@@ -279,59 +274,46 @@ class ChromedashFeaturelist extends LitElement {
     return null;
   }
 
-  scrollToMilestone(milestone) {
-    const id = this._firstOfMilestone(milestone);
-    if (id) {
-      this.scrollToId(id);
-    }
-  }
-
   // Directly called from template/features.html
   scrollToId(targetId) {
     if (!targetId) return;
 
-    let selector;
-    if (!isNaN(Number(targetId))) {
-      /* number is not valid id. so we added prefix 'id-' in the template. */
-      selector = `#id-${targetId}`;
-    } else {
-      selector = `#${targetId}`;
+    for (let i = 0, f; f = this.filtered[i]; ++i) {
+      if (f.id === targetId) {
+        this.virtualizerEl.scrollToIndex(i);
+        const featureEl = document.querySelector('chromedash-featurelist').shadowRoot.querySelector(`chromedash-feature[id=id-${f.id}]`);
```

Comment:
I think this is fine to get things working, but it does defeat the encapsulation of shadow dom.

Perhaps a cleaner model would be to set the id on chromedash-featurelist and have it locate the correct child and set the open property on it. Ideally this should all be done through the public interfaces of these components without needing to query into their shadow roots.

Reply:
+1

At least add a comment before this line saying why you're doing this and a TODO to fix it.

Reply:
Thanks for the suggestion. Applied the new method.